File: app/src/main/java/com/fernandosierra/door2door/di/AppComponent.java
package com.fernandosierra.door2door.di;

import android.app.Application;

import com.fernandosierra.door2door.Door2DoorApp;
import com.fernandosierra.door2door.data.di.DataModule;
import com.fernandosierra.door2door.presentation.di.PresentationModule;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;

@Component(modules = {
        AppModule.class,
        AndroidInjectionModule.class,
        ActivityBuilder.class,
        PresentationModule.class,
        DataModule.class
})
@Singleton
public interface AppComponent {

    void inject(Door2DoorApp app);

    @Component.Builder
    interface Builder {

        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }
}

File: app/src/main/java/com/fernandosierra/door2door/di/AppComponent.kt
package com.fernandosierra.door2door.di

import android.app.Application
import com.fernandosierra.door2door.Door2DoorApp
import com.fernandosierra.door2door.data.di.DataModule
import com.fernandosierra.door2door.presentation.di.PresentationModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import javax.inject.Singleton

@Component(modules = [(AppModule::class),
    (AndroidInjectionModule::class),
    (ActivityBuilder::class),
    (PresentationModule::class),
    (DataModule::class)])
@Singleton
interface AppComponent {

    fun inject(app: Door2DoorApp)

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }
}